drivers/SmartThings/harman-luxury/src/devices.lua

```diff
@@ -0,0 +1,142 @@
+
+local devices_SetupID = {
+    L75ms = "602",
+    L42ms = "603",
+    AVR5 = "604"
+}
+
+function Devices.GetSupportedDevices()
+    log.info("GetSupportedDevices: supported models: " .. st_utils.stringify_table(suppourted_devices))
+    return suppourted_devices
+end
+
+local function GetDefaultDeviceInfo(dni, ip)
+    local label = api.GetDeviceName(ip)
+    if not label then
+        log.warn("Failed to get Device Name from device with IP:" .. ip)
+        label = DEFAULT_DEVICE_NAME
+    end
+    local manufacturer = api.GetManufatureName(ip)
+    if not manufacturer then
+        log.warn("Failed to get Manufacture Name from device with IP:" .. ip)
+        manufacturer = DEFAULT_MANUFACTURE_NAME
+    end
+    local model = api.GetModelName(ip)
+    if not model then
+        log.warn("Failed to get Device Name from device with IP:" .. ip)
+        model = DEFAULT_MODEL_NAME
+    end
+    local vendor = api.GetProductName(ip)
+    if not vendor then
+        log.warn("Failed to get Product Name from device with IP:" .. ip)
+        vendor = DEFAULT_PRODUCT_NAME
+    end
+
+    local device_info = {
+        type = "LAN",
+        device_network_id = dni,
+        label = label,
+        profile = "harman-luxury",
+        manufacturer = manufacturer,
+        model = model,
+        vendor_provided_label = vendor
+    }
+
+    return device_info
+end
+
+local function GetL75msDeviceInfo(dni, ip)
+    local label = api.GetDeviceName(ip)
+    if not label then
+        log.warn("Failed to get Device Name from device with IP:" .. ip)
+        label = DEFAULT_DEVICE_NAME
+    end
+
+    local device_info = {
+        type = "LAN",
+        device_network_id = dni,
+        label = label,
+        profile = "harman-luxury",
+        manufacturer = "JBL",
+        model = "L75ms",
+        vendor_provided_label = label
+    }
+
+    return device_info
+end
+
+local function GetL42msDeviceInfo(dni, ip)
+    local label = api.GetDeviceName(ip)
+    if not label then
+        log.warn("Failed to get Device Name from device with IP:" .. ip)
+        label = DEFAULT_DEVICE_NAME
+    end
+
+    local device_info = {
+        type = "LAN",
+        device_network_id = dni,
+        label = label,
+        profile = "harman-luxury",
+        manufacturer = "JBL",
+        model = "L42ms",
+        vendor_provided_label = label
+    }
+
+    return device_info
+end
+
+local function GetAvr5DeviceInfo(dni, ip)
+    local label = api.GetDeviceName(ip)
+    if not label then
+        log.warn("Failed to get Device Name from device with IP:" .. ip)
+        label = DEFAULT_DEVICE_NAME
+    end
+
+    local device_info = {
+        type = "LAN",
+        device_network_id = dni,
+        label = label,
+        profile = "harman-luxury",
+        manufacturer = "ARCAM",
+        model = "AVR5",
+        vendor_provided_label = label
+    }
+
+    return device_info
+end
+
+function Devices.get_device_info(dni, params)
```

Comment:
Similar to the comment that @greens made elsewhere, there's a lot of duplicated code here. One option would be to have Lua tables that take `params.setupid` as the key, and return the MFR and Model information for the various speakers, as those are the only things that differ in the various functions defined in this file.